lms/djangoapps/commerce/api/v1/models.py

```diff
@@ -22,6 +22,19 @@ def __init__(self, id, modes):  # pylint: disable=invalid-name,redefined-builtin
         self.modes = list(modes)
         self._deleted_modes = []
 
+    @property
+    def name(self):
+        """ Return course name. """
+        course_id = CourseKey.from_string(unicode(self.id))  # pylint: disable=invalid-name
+
+        try:
+            return CourseOverview.get_from_id(course_id).display_name
+        except CourseOverview.DoesNotExist:
+            # NOTE (CCB): Ideally, the course modes table should only contain data for courses that exist in
+            # modulestore. If that is not the case, say for local development/testing, carry on without failure.
+            log.warning('Failed to retrieve CourseOverview for [%s]. Using empty course name.', course_id)
+            return None
```

Comment:
Yes, an empty name will break course product creation. Ironically, it is Otto that creates these invalid course modes. I noticed during the demo on Friday that folks were creating data for non-existent courses.

We are not presently using this API to retrieve course names; however, if we do, Otto needs to be updated to account for this invalid data (which should only exist on stage).

I created XCOM-538 to update the CAT and Commerce API.